### PR TITLE
NPC Addition: The Hermit

### PR DESCRIPTION
**Pull request by chaosvolt**
This is formerly dependent on https://github.com/CleverRaven/Cataclysm-DDA/pull/27406, now dependent on https://github.com/CleverRaven/Cataclysm-DDA/pull/27453, and has a couple things I still plan to add, but now it's at a state where I can self-PR for testing and feedback.

The big thing added is a location you can run into having an NPC, who unlocks a quest line that, if followed, can guide the player down the path of becoming a Paragon of The Veil. Getting him to open up requires a bit of convincing, with several options depending on your profession and other conditions.

In addition this also sees the use of mission stuff to generate a couple locations on demand rather than pre-generating them, profession traits for arcanist characters, a few assorted tweaks here and there.

Finally, the fatigue cost of Veil spells now differ from versions obtained by other means. This is a practical benefit added to some mutation changes needed to make the third quest work without letting the player "cheat" by obtaining spells via spellcraft scrolls or other means.

Todo:
1. Add a method that allows completing the second quest with an illiterate character. Will wait until the ability to hand over the sword and other materials exists in such a way that it doesn't dupe the cursed sword is implemented.
2. Add a failure check preceding the fourth quest for if the character has a mutation threshold, directing them to find another. Can do it now (Mark's PR supporting it is up) but I need to figure out the specifics, which are closely related to point 4 and 5.
3. Add failure condition for the fourth quest while it's in progress, further adding to the "find another" plot hook.
4. Instance effect checks in such a way that the hermit has an understanding of overall quest progression (NPC effects), while tracking whether individual players are trusted (player effects).
5. Allow the same current outcome (another character can pick up where the last one left off) while still requiring that character to earn the hermit's trust, and additionally add some way to get the quest chain moving again if the previous character died with their quest active (rendering the quest chain unsolvable).
6. Add a post-fourth-quest side quest where the player has the option of bringing the hermit materials to copy Oaths to The Chalice. Will likely entail an effect marking the hermit as busy copying for several days (20? 24? 28? 30?) and receiving a copy of the book afterward. Mostly a flavor thing unless expanded on by recruiting NPCs or other future quest shenanigans, hence not bothering with it now.
7. Add a resolution involving skipping all this quest nonsense if you show up as a Paragon of The Veil right off the bat.

As usual, big thanks to @mlangsdorf for all the work that's made all of this possible.

### CODE REVIEW

File: Arcana/TALK_HERMIT.json [hunk 1]
```diff
@@ -0,0 +1,932 @@
...
+      }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_HERMIT",
+    "dynamic_line": {
+      "npc_has_effect": "hermit_earned_trust_1",
+      "yes": {
+        "u_has_any_trait": [ "THRESH_VEIL" ],
```
Comment (mlangsdorf) [inline Arcana/TALK_HERMIT.json]:
```suggestion
        "u_has_trait": "THRESH_VEIL",
```

Comment (chaosvolt) [reply Arcana/TALK_HERMIT.json]: Ah, nice. Force of habit it seems, as `has_any_trait` is so old (https://github.com/CleverRaven/Cataclysm-DDA/pull/13867), and I seem to have not noticed that you added a version for single traits. Saves a tiny bit of characters, but useful to know, thank you.

And damn, even getting through that much of it is astounding, sorry that I went a fair bit overboard.

**Comment by chaosvolt** [comment]
Now it's officially in a state where I'm happy with it, once https://github.com/CleverRaven/Cataclysm-DDA/pull/27453 is merged I'll go ahead with my own.

**Comment by chaosvolt** [comment]
Well, minor complication: `u_has_item` can't sanely handle items held in a scabbard/holster OR wielded, so this will stay on hold until a way around that is fixed...

**Pull request merged**